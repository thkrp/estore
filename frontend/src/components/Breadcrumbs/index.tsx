import React, { useEffect, useState } from 'react';
import { ReactComponent as LoaderIcon } from '../../assets/icons/horizontal-loader.svg';
import {
    CategoriesStyled,
    ElementStyled,
    LinkStyled,
    ListStyled,
    MoreCategoriesButtonStyled,
    WrapperStyled
} from './index.styles';
import useBreadcrumbs from '../../common/hooks/use.breadcrumbs';
import { transformToKey } from '../../common/helpers/key.helper';

const Breadcrumbs = () => {
    const breadcrumbs = useBreadcrumbs();
    const [showCategories, setShowCategories] = useState<Record<string, boolean>>({});

    useEffect(() => {
        setShowCategories({});
    }, [breadcrumbs]);

    return (
        <WrapperStyled>
            <ListStyled role="navigation" aria-label="breadcrumbs">
                {breadcrumbs.length
                    ? breadcrumbs.map((el, index) =>
                          index === breadcrumbs.length - 1 ? (
                              <ElementStyled key={transformToKey(index, el.code)}>
                                  {el.name || <LoaderIcon height="15px" role="img" />}
                              </ElementStyled>
                          ) : (
                              <ElementStyled key={transformToKey(index, el.code)}>
                                  <LinkStyled to={el.url}>{el.name || el.code}</LinkStyled>
                                  {el.children && (
                                      <>
                                          {breadcrumbs.length - 1 !== index && (
                                              <MoreCategoriesButtonStyled
                                                  role="presentation"
                                                  onClick={() =>
                                                      setShowCategories(prev => ({
                                                          [el.code]: !prev[el.code]
                                                      }))
                                                  }
                                              />
                                          )}
                                          {showCategories[el.code] && (
                                              <CategoriesStyled role="menu" aria-label="sub-categories">
                                                  {el.children.map((cat, i) => (
                                                      <LinkStyled
                                                          to={cat.url}
                                                          key={transformToKey(i, `${cat.url}`)}
                                                          role="menuitem"
                                                      >
                                                          {cat.name || cat.code}
                                                      </LinkStyled>
                                                  ))}
                                              </CategoriesStyled>
                                          )}
                                      </>
                                  )}
                              </ElementStyled>
                          )
                      )
                    : null}
            </ListStyled>
        </WrapperStyled>
    );
};

export default Breadcrumbs;
