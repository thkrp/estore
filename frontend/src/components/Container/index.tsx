import React, { FC, ReactNode } from 'react';
import { WrapperStyled } from './index.styles';

type Props = {
    children: ReactNode;
    className?: string;
};
/**
 * A wrapper for content that limits the width.
 */
const Container: FC<Props> = ({ children, className }) => (
    <WrapperStyled className={className}>{children}</WrapperStyled>
);

export default Container;
