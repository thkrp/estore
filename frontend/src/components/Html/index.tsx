import React from 'react';
import DOMPurify from 'dompurify';

type Props = {
    dirty: string;
};
const CleanHtml = ({ dirty }: Props) => {
    const clean = DOMPurify.sanitize(dirty);
    // eslint-disable-next-line react/no-danger
    return <div dangerouslySetInnerHTML={{ __html: clean }} />;
};

export default CleanHtml;
