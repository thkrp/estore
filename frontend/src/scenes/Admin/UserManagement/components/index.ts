import ActionBar from './ActionBar';
import UserList from './UserList';

export { UserList, ActionBar };
