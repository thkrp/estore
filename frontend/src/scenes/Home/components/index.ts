import BestSales from './BestSales';
import Brands from './Brands';
import Tabs from './Tabs';
import ViewedProducts from './ViewedProducts';

export { Tabs, BestSales, ViewedProducts, Brands };
