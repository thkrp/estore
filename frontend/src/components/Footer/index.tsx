import React, { FC, ReactElement } from 'react';
import { BottomMenu, BottomMenuItemDto } from 'app-shared';
import {
    FooterStyled,
    ListHeadingStyled,
    ListStyled,
    ListItemStyled,
    ListWrapperStyled,
    LogoDescriptionStyled,
    LogoWrapperStyled,
    MenuWrapperStyled,
    WrapperStyled
} from './index.styles';
import Container from '../Container';
import { FooterInfo } from '../../common/types/footer.info';
import Logo from '../Logo';
import Link from '../Links/Link';
import ContactLink from '../Links/ContactLink';
import ExternalLink from '../Links/ExternalLink';

type Props = {
    info?: FooterInfo;
    menu?: BottomMenu[];
};

enum LinkTypes {
    section = 'SECTION',
    link = 'LINK',
    email = 'EMAIL',
    phone = 'PHONE',
    external = 'EXTERNAL'
}

const Links: { [key in LinkTypes]: (item: BottomMenuItemDto) => ReactElement } = {
    [LinkTypes.section]: ({ url, name }) => <Link to={url}>{name}</Link>,
    [LinkTypes.link]: ({ url, name }) => <Link to={url}>{name}</Link>,
    [LinkTypes.phone]: ({ name }) => <ContactLink value={name} type="phone" hiddenIcon />,
    [LinkTypes.email]: ({ name }) => <ContactLink value={name} type="email" hiddenIcon />,
    [LinkTypes.external]: ({ url, name }) => <ExternalLink href={url} title={name} />
};

const Footer: FC<Props> = ({ menu, info }) => {
    return (
        <WrapperStyled>
            <Container>
                <FooterStyled>
                    <LogoWrapperStyled>
                        <Logo logo={info?.logo_footer} />
                        <LogoDescriptionStyled>{info?.footer_description}</LogoDescriptionStyled>
                    </LogoWrapperStyled>
                    <MenuWrapperStyled>
                        {!!menu?.length &&
                            menu.map(list => (
                                <ListWrapperStyled key={list.name}>
                                    <ListHeadingStyled>{list.name}</ListHeadingStyled>
                                    <ListStyled>
                                        {list.items.map(li => {
                                            const LinkComponent = Links[li.type as LinkTypes];
                                            return <ListItemStyled key={li.name}>{LinkComponent(li)}</ListItemStyled>;
                                        })}
                                    </ListStyled>
                                </ListWrapperStyled>
                            ))}
                    </MenuWrapperStyled>
                </FooterStyled>
            </Container>
        </WrapperStyled>
    );
};

export default Footer;
