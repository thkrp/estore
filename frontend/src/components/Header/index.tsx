import React, { FC } from 'react';
import { GeneralInfo, TopMenu } from 'app-shared';
import TopNav from '../Navigation/Top';
import { HeaderStyled } from './index.styles';
import Container from '../Container';
import Logo from '../Logo';
import Actions from './Actions';
import Cart from './Cart';

type Props = {
    menu?: TopMenu;
    info?: GeneralInfo;
};

const Header: FC<Props> = ({ menu, info }) => {
    return (
        <header>
            <TopNav menu={menu} />
            <Container>
                <HeaderStyled>
                    <Logo logo={info && info.logo} />
                    <Actions phones={info?.phones} emails={info?.emails} />
                    <Cart />
                </HeaderStyled>
            </Container>
        </header>
    );
};

export default Header;
