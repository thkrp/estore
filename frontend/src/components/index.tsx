import Container from './Container';
import Header from './Header';
import Footer from './Footer';
import CatalogNav from './Navigation/Catalog';
import TopNav from './Navigation/Top';
import Breadcrumbs from './Breadcrumbs';
import Link from './Links/Link';
import HorizontalSlider from './Sliders/Horizontal';
import BrandCard from './Cards/BrandCard';
import ProductCard from './Cards/ProductCard';
import SpecialCategories from './SpecialCategories';
import HomeSlider from './Sliders/Home';
import InputText from './InputText';
import MobileNav from './Navigation/Mobile';
import Search from './Search';
import CleanHtml from './Html';
import Login from './MaterialUi/Auth/Login';

export {
    Container,
    Header,
    Footer,
    CatalogNav,
    TopNav,
    Breadcrumbs,
    Link,
    HorizontalSlider,
    ProductCard,
    BrandCard,
    SpecialCategories,
    HomeSlider,
    InputText,
    MobileNav,
    Search,
    CleanHtml,
    Login
};
