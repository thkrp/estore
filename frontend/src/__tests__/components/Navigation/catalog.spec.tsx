/* eslint-disable no-restricted-imports*/
import React from 'react';
import '@testing-library/jest-dom';
import { cleanup, render } from '@testing-library/react';
import { Locale, Localization, mockData } from 'app-shared';
import { MemoryRouter } from 'react-router-dom';
import * as sanitizer from '@braintree/sanitize-url';
import { IntlProvider } from 'react-intl';
import { CatalogNav } from '../../../components';

describe('CatalogNav', () => {
    afterEach(cleanup);

    it('should be defined', () => {
        const { container } = render(
            <IntlProvider key={Localization.ru} locale={Locale.ru} messages={mockData.localizationData}>
                <CatalogNav menu={mockData.catalogMenu} />
            </IntlProvider>,
            { wrapper: MemoryRouter }
        );
        expect(container).toBeDefined();
        expect(container).toMatchSnapshot();
    });

    it('sanitizeUrl has been called', () => {
        jest.spyOn(sanitizer, 'sanitizeUrl');
        render(
            <IntlProvider key={Localization.ru} locale={Locale.ru} messages={mockData.localizationData}>
                <CatalogNav menu={mockData.catalogMenu} />
            </IntlProvider>,
            { wrapper: MemoryRouter }
        );
        expect(sanitizer.sanitizeUrl).toHaveBeenCalled();
    });

    it('should return null if no menu', () => {
        const { container } = render(<CatalogNav />, { wrapper: MemoryRouter });
        expect(container).toBeEmptyDOMElement();
    });
});
