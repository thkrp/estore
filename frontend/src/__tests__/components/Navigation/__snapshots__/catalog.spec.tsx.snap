// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogNav should be defined 1`] = `
<div>
  <div
    class="sc-ilxebA hQzSdn"
  >
    <div
      class="sc-jIBlqr iKkUiM"
    >
      <span
        class="sc-hhyKGa ilIzZR"
      >
        <svg>
          list.svg
        </svg>
      </span>
      <a
        class="sc-gYrqIg hZxwYq"
        href="/catalog"
      >
        КАТАЛОГ ТОВАРОВ
      </a>
    </div>
    <div
      class="sc-cCzLxZ bYuvCp"
    >
      <a
        class="sc-ckdEwu fugiOn"
        href="/catalog/tractors"
        title="Тракторы"
      >
        Тракторы
      </a>
      <a
        class="sc-ckdEwu fugiOn"
        href="/catalog/motobloki"
        title="Мотоблоки"
      >
        Мотоблоки
      </a>
      <a
        class="sc-ckdEwu fugiOn"
        href="/catalog/traktornoe-navesnoe-oborudovanie"
        title="Тракторное навесное оборудование"
      >
        Тракторное навесное оборудование
      </a>
      <a
        class="sc-ckdEwu fugiOn"
        href="/catalog/motoblochnoe-navesnoe-oborudovanie"
        title="Мотоблочное навесное оборудование"
      >
        Мотоблочное навесное оборудование
      </a>
      <a
        class="sc-ckdEwu fugiOn"
        href="/catalog/sredstva-zashchity-rasteniy"
        title="Средства защиты растений"
      >
        Средства защиты растений
      </a>
      <a
        class="sc-ckdEwu fugiOn"
        href="/catalog/tekhnika-dlya-sada-i-ogoroda"
        title="Техника для сада и огорода"
      >
        Техника для сада и огорода
      </a>
    </div>
  </div>
</div>
`;
